dartagnan/src/test/java/com/dat3m/dartagnan/parsers/program/ParserSpirvTest.java

```diff
@@ -0,0 +1,54 @@
+package com.dat3m.dartagnan.parsers.program;
+
+import com.dat3m.dartagnan.program.Program;
+import org.antlr.v4.runtime.CharStream;
+import org.antlr.v4.runtime.CharStreams;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.stream.Stream;
+
+import static com.dat3m.dartagnan.utils.ResourceHelper.getTestResourcePath;
+import static org.junit.Assert.assertNotNull;
+
+@RunWith(Parameterized.class)
+public class ParserSpirvTest {
+
+    private static final String PATH = "parsers/program/spirv/";
+
+    private final String filename;
+
+    public ParserSpirvTest(String filename) {
+        this.filename = filename;
+    }
+
+    @Parameterized.Parameters(name = "{index}: {0}")
+    public static Iterable<Object[]> data() throws IOException {
+        String path = Paths.get(getTestResourcePath(PATH)).toString();
+        File dir = new File(path);
+        File[] files = dir.listFiles();
+        assertNotNull(files);
+        return Stream.of(files)
+                .filter(file -> !file.isDirectory())
+                .map(f -> new Object[]{f.getName()})
+                .toList();
+    }
+
+    @Test
+    public void testParsingFile() throws IOException {
+        Program program;
+        Path path = Paths.get(getTestResourcePath("parsers/program/spirv/" + filename));
```

Comment:
You can use `PATH` variable

Reply:
I think "getTestResourcePart" is the standard way to get a path to resource file. Relying on PATH or other env variables is 1) not a standard practice 2) is fragile and can break with different OS or configurations.

Reply:
I was not referring to replace `getTestResourcePart`, but rather the string that is passed as the first part of its argument. I.e., similar to line 33

Reply:
Ah, then you are right. Done.